Fix classical control rendering for gate

gadhvirushiraj opened the pull request:
- Fixed rendering issue for classical control on gates for both renderers.
- Optimize: Removed `_draw_singleq_gate` for rendering in MatRenderer.

BoxiLi commented:
Thanks a lot for such a quick PR! 

I noticed two things: 
1. This classical control line should also be added when we compute the layer of the gate. 
```python
from qutip_qip.circuit import QubitCircuit
qc = QubitCircuit(2, num_cbits=2)
qc.add_gate("X", targets=0, classical_controls=0, classical_control_value=0)
qc.add_gate("X", targets=1, classical_controls=1, classical_control_value=1)
qc.draw()
```
![image](https://github.com/user-attachments/assets/f3e6c116-648b-4f41-b3d1-06360dd620b0)

2. I think there might be a misunderstanding of the `classical_controls` and `classical_control_value`. I didn't realize it when you asked that question on Discord. I think here we should read `classical_controls`, which should be a list that records the control classical bits. And then we draw a circuit on it, like in the quantum controlled gate

![image](https://github.com/user-attachments/assets/7853b3d0-bcf4-4b40-b5f7-a6e06b6f6509)

gadhvirushiraj commented:
> 1. This classical control line should also be added when we compute the layer of the gate.

We dont maintain a record for classical lines. But we just augment this by maintaining gaps in qlines. But it should be now fixed for MatRenderer.

I have modified it to read `classical_controls` too.

BoxiLi commented:
Great! Looks good!

BoxiLi commented:
Looks good! Rendered nicely on my machine. Would be nice to add one test case for the text renderer. Just to be safe.

The failing test is unrelated, but the one on matplotlib seems to be a warning from the old matplotlib 3.9

gadhvirushiraj commented:
@BoxiLi, Sorry for the delay.
Yes, please merge - the failed test looks unrelated.
